blocks/subscriptions-block/components/ContactInfo/index.jsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Button, BUTTON_SIZES, BUTTON_STYLES, FIELD_TYPES, FormInputField, PrimaryFont,
+} from '@wpmedia/shared-styles';
+import React, { useRef } from 'react';
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+import getTranslatedPhrases from 'fusion:intl';
+import countryCodes from './countryCodes';
+import GoogleIcon from './google.svg';
+import FacebookIcon from './facebook.svg';
+import './styles.scss';
+
+const ContactInfo = ({ callback, user, logoutCallback }) => {
+  const formRef = useRef();
+  const entriesRef = useRef({});
+
+  if (user) {
+    entriesRef.current.email = user.email ? user.email : '';
+    entriesRef.current.firstName = user.firstName ? user.firstName : '';
+    entriesRef.current.lastName = user.lastName ? user.lastName : '';
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const valid = formRef.current.checkValidity();
+    if (valid) {
+      callback(entriesRef.current);
+    }
+  };
+
+  const handleInputChange = (name, entry) => {
+    entriesRef.current[name] = entry.value;
+  };
+
+  const handleLogout = () => {
+    logoutCallback();
```

Comment:
when I clicked this as a google signed in user, the email and first and last name persisted. is that expected?